tests/Domain/Integrations/Controllers/IntegrationControllerTest.php

```diff
@@ -88,11 +207,447 @@ public function test_it_can_active_an_integration_with_an_organization(): void
+
+        $this->assertDatabaseHas('contacts', [
+            'id' => $functionalContact->id->toString(),
+            'integration_id' => $integration->id->toString(),
+            'email' => 'other@test.com',
+            'type' => $functionalContact->type->value,
+            'first_name' => $functionalContact->firstName,
+            'last_name' => $functionalContact->lastName,
+        ]);
+    }
+
+    public function test_it_cant_update_contacts_if_unauthorized(): void
+    {
+        $this->actingAs(UserModel::createSystemUser(), 'web');
+
+        $integration = $this->givenThereIsAnIntegration();
+        $functionalContact = $this->givenThereIsAFunctionalContactOnIntegration($integration);
+
+        $response = $this->patch(
+            "/integrations/{$integration->id}/contacts",
+            [
+                'functional' => [
+                    'id' => $functionalContact->id->toString(),
+                    'integrationId' => $integration->id->toString(),
+                    'email' => 'other@test.com',
+                    'type' => $functionalContact->type->value,
+                    'firstName' => $functionalContact->firstName,
+                    'lastName' => $functionalContact->lastName,
+                ],
+            ]
+        );
+
+        $response->assertForbidden();
+
+        $this->assertDatabaseHas('contacts', [
+            'id' => $functionalContact->id->toString(),
+            'integration_id' => $integration->id->toString(),
+            'email' => $functionalContact->email,
+            'type' => $functionalContact->type->value,
+            'first_name' => $functionalContact->firstName,
+            'last_name' => $functionalContact->lastName,
+        ]);
+    }
+
+    public function test_it_can_destroy_a_contact(): void
+    {
+        $this->actingAs(UserModel::createSystemUser(), 'web');
+
+        $integration = $this->givenThereIsAnIntegration();
+        $this->givenTheActingUserIsAContactOnIntegration($integration);
+        $functionalContact = $this->givenThereIsAFunctionalContactOnIntegration($integration);
+
+        $this->delete("/integrations/{$integration->id}/contacts/{$functionalContact->id}");
+
+        $this->assertSoftDeleted('contacts', [
+            'id' => $functionalContact->id->toString(),
+        ]);
+    }
+
+    public function test_it_cant_destroy_a_contact_if_unauthorized(): void
+    {
+        $this->actingAs(UserModel::createSystemUser(), 'web');
+
+        $integration = $this->givenThereIsAnIntegration();
+        $functionalContact = $this->givenThereIsAFunctionalContactOnIntegration($integration);
+
+        $response = $this->delete("/integrations/{$integration->id}/contacts/{$functionalContact->id}");
+
+        $response->assertForbidden();
+
+        $this->assertNotSoftDeleted('contacts', [
+            'id' => $functionalContact->id->toString(),
+        ]);
+    }
+
+    public function test_it_can_update_billing_info_of_organization(): void
```

Comment:
The term `billing_info` can be removed there is only the concept of an `organization`. The term `billing_info` came from a UI view but got fully removed from the domain, see https://github.com/cultuurnet/publiq-platform/pull/665

Reply:
there is are still remaining billing info endpoints:

`Route::patch('/integrations/{id}/billing', [IntegrationController::class, 'updateBilling']);`
In the integration controller the request handler is still called `updateBilling`.
Should I change it to update organisation?

Reply:
Done